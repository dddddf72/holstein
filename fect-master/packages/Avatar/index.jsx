import { computed, toRefs } from 'vue'
import { validator, theme, createNameSpace } from '../utils'
const [createComponent] = createNameSpace('Avatar')
const { normalSizes } = theme

import './avatar.less'

export default createComponent({
  props: {
    stacked: Boolean,
    isSquare: Boolean,
    size: {
      type: String,
      validator: validator.enums(normalSizes),
      default: 'medium',
    },
    text: {
      type: String,
      default: '',
    },
    src: String,
    className: String,
  },
  setup(props, { attrs }) {
    const { stacked, isSquare, size, text, src, className } = toRefs(props)
    const showText = !(src && src.value)

    // when text has value to take the best range
    const safeText = (text) =>
      String(text).length <= 4 ? text : text.slice(0, 3)

    //calculate outer Container className
    const calcClass = computed(() => (className ? className.value : ''))

    // claulate avatar style
    const calcAttrs = computed(() => {
      let str = ''
      stacked.value && (str += ' stacked')
      isSquare.value && (str += ' isSquare')
      size.value && (str += ` ${size.value}`)
      return str.trim()
    })

    return () => (
      <>
        <div
          className={`fect-avatar ${calcAttrs.value} ${calcClass.value}`}
          style={attrs.style ? attrs.style : ''}
        >
          {!showText && <img src={src.value} draggable="false" {...attrs} />}
          {showText && (
            <span
              className={`fect-avatar-text ${attrs.class ? attrs.class : ''}`}
              {...attrs}
            >
              {/* {text.value} */}
              {safeText(text.value)}
            </span>
          )}
        </div>
      </>
    )
  },
})
