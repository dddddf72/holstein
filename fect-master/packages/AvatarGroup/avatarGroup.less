.fect-ava-group {
  display: flex;
  align-items: center;
  height: auto;
  width: max-content;
}

.fect-ava-counter {
  font-size: 0.825em;
  display: inline-flex;
  align-items: center;
  padding-left: 5px;
  color: var(--accents-7);
}
