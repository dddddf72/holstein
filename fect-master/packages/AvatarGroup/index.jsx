import { toRefs } from 'vue'
import { createNameSpace } from '../utils'

const [createComponent] = createNameSpace('AvatarGroup')

import './avatarGroup.less'

export default createComponent({
  props: {
    count: [String, Number],
  },
  setup(props, { attrs, slots }) {
    const { count } = toRefs(props)
    const showCount = !!(count && count.value)

    return () => (
      <>
        <div
          {...attrs}
          className={`fect-ava-group ${(attrs.class
            ? attrs.class
            : ''
          ).trim()}`}
          style={attrs.style}
        >
          {slots.default?.()}
          {showCount && (
            <span className={'fect-ava-counter'}>+{count.value}</span>
          )}
        </div>
      </>
    )
  },
})
