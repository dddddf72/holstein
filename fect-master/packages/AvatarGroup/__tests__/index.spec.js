import { mount } from '@vue/test-utils'
import AvatarGroup from '../index'
import Avatar from '../../Avatar'

const NUMBER_COUNT = 10

describe('AvatarGroup', () => {
  it('should be render as Element', () => {
    const group = mount(AvatarGroup, {})
    expect(group.html()).toMatchSnapshot()
  })

  it('shoudle be avatar container and should work', () => {
    const wrapper = mount({
      setup(props) {
        return {
          list: ['a', 'b', 'c', 'd'],
        }
      },
      render() {
        return (
          <AvatarGroup count={NUMBER_COUNT}>
            {this.list.map((item) => (
              <Avatar text={item} stacked></Avatar>
            ))}
          </AvatarGroup>
        )
      },
    })

    expect(wrapper.html()).toMatchSnapshot()

    expect(wrapper.find('.fect-ava-counter').text()).toBe('+10')

    expect(() => wrapper.unmount()).not.toThrow()
  })
})
