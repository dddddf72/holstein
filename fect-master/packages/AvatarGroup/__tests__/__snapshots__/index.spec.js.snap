// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup shoudle be avatar container and should work 1`] = `
<div class="fect-ava-group ">
  <div class="fect-avatar stacked medium ">
    <!----><span class="fect-avatar-text ">a</span>
  </div>
  <div class="fect-avatar stacked medium ">
    <!----><span class="fect-avatar-text ">b</span>
  </div>
  <div class="fect-avatar stacked medium ">
    <!----><span class="fect-avatar-text ">c</span>
  </div>
  <div class="fect-avatar stacked medium ">
    <!----><span class="fect-avatar-text ">d</span>
  </div><span class="fect-ava-counter">+10</span>
</div>
`;

exports[`AvatarGroup should be render as Element 1`] = `
<div class="fect-ava-group ">
  <!---->
  <!---->
</div>
`;
